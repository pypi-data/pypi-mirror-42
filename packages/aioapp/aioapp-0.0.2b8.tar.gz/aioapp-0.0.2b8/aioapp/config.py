import copy
import os
from collections import OrderedDict
from typing import Optional, Union, Dict, Any, Type
from os import _Environ

Env = Union[_Environ, dict]


class ConfigError(Exception):
    pass


class Val:

    def __init__(self, name: str, value: Any) -> None:
        self.name = name
        self.value = value

    def __call__(self):  # pragma: nocover
        raise NotImplementedError()

    @staticmethod
    def type_name() -> str:
        return ''

    def args_markdown(self) -> str:
        return ''


class StrVal(Val):

    def __init__(self, name: str, value: Any,
                 max: Optional[int] = None,
                 min: Optional[int] = None) -> None:
        super().__init__(name, value)
        self.min = min
        self.max = max

    def __call__(self) -> str:
        if not isinstance(self.value, str):
            raise ConfigError("%s must be a string" % self.name)
        if self.min is not None and len(self.value) < self.min:
            raise ConfigError("length of %s must be greater than or equal to "
                              "%s"
                              "" % (self.name, self.min))
        if self.max is not None and len(self.value) > self.max:
            raise ConfigError("length of %s must be less than or equal to %s"
                              "" % (self.name, self.max))

        return self.value

    @staticmethod
    def type_name() -> str:
        return 'string'

    def args_markdown(self) -> str:
        text = ''
        if self.min is not None:
            text += '\n  min length: %s' % self.min
        if self.max is not None:
            text += '\n  max length: %s' % self.max
        return text


class BoolVal(Val):

    def __call__(self) -> bool:
        if isinstance(self.value, bool):
            return self.value
        if isinstance(self.value, int):
            return self.value != 0
        if isinstance(self.value, str):
            if self.value.lower() in ('1', 'on', 'true', 't'):
                return True
            if self.value.lower() in ('0', 'off', 'false', 'f'):
                return False
        raise ConfigError("%s must be a boolean" % self.name)

    @staticmethod
    def type_name() -> str:
        return 'boolean'


class IntVal(Val):

    def __init__(self, name: str, value: Any,
                 max: Optional[int] = None,
                 min: Optional[int] = None) -> None:
        super().__init__(name, value)
        self.min = min
        self.max = max

    def __call__(self) -> int:
        try:
            val = int(self.value)
        except Exception:
            raise ConfigError("%s must be an integer" % self.name)
        if self.min is not None and val < self.min:
            raise ConfigError("%s must be greater than or equal to %s"
                              "" % (self.name, self.min))
        if self.max is not None and val > self.max:
            raise ConfigError("%s must be less than or equal to %s"
                              "" % (self.name, self.max))
        return val

    @staticmethod
    def type_name() -> str:
        return 'integer'

    def args_markdown(self) -> str:
        text = ''
        if self.min is not None:
            text += '\n  min value: %s' % self.min
        if self.max is not None:
            text += '\n  max value: %s' % self.max
        return text


class FloatVal(Val):

    def __init__(self, name: str, value: Any,
                 max: Optional[float] = None,
                 min: Optional[float] = None) -> None:
        super().__init__(name, value)
        self.min = min
        self.max = max

    def __call__(self) -> float:
        try:
            val = float(self.value)
        except Exception:
            raise ConfigError("%s must be a float" % self.name)
        if self.min is not None and val < self.min:
            raise ConfigError("%s must be greater than or equal to %s"
                              "" % (self.name, self.min))
        if self.max is not None and val > self.max:
            raise ConfigError("%s must be less than or equal to %s"
                              "" % (self.name, self.max))
        return val

    @staticmethod
    def type_name() -> str:
        return 'float'

    def args_markdown(self) -> str:
        text = ''
        if self.min is not None:
            text += '\n  min value: %s' % self.min
        if self.max is not None:
            text += '\n  max value: %s' % self.max
        return text


class FileVal(Val):

    def __init__(self, name: str, value: Any,
                 mode: str = 'r',
                 encoding: str = 'UTF-8') -> None:
        super().__init__(name, value)
        self.mode = mode
        self.encoding = encoding

    def __call__(self, mode: str = 'r', encoding: str = 'UTF-8') -> str:
        try:
            with open(self.value, mode, encoding=encoding):
                pass
        except Exception as e:
            raise ConfigError("Could not access to file %s with error: %s"
                              "" % (self.value, e))
        return self.value

    @staticmethod
    def type_name() -> str:
        return 'string(path to file)'

    def args_markdown(self) -> str:
        text = ''
        if self.mode is not None:
            text += '\n  assess mode: %s' % self.mode
        if self.encoding is not None:
            text += '\n  encoding: %s' % self.encoding
        return text


class DirVal(Val):

    def __call__(self) -> str:
        if not os.path.exists(self.value):
            raise ConfigError("Directory %s does not exist"
                              "" % self.value)
        return self.value

    @staticmethod
    def type_name() -> str:
        return 'string(path to dir)'


class Config:
    _vars: Dict[str, Union[Dict, OrderedDict]] = {}

    def __init__(self,
                 env: Optional[Env] = None) -> None:
        self._env = env or os.environ
        self._conf = copy.deepcopy(self._vars)
        self._description: Dict[str, Dict] = {}
        for key, val in self._conf.items():
            val_type = val.pop('type')
            val_name = val.pop('name')
            val_default = None
            if 'default' in val:
                val_default = val.pop('default')
            required = False
            if 'required' in val:
                required = bool(val.pop('required'))
            descr = ''
            if 'descr' in val:
                descr = str(val.pop('descr'))

            self._description[val_name] = {
                'type': val_type,
                'default': val_default,
                'required': required,
                'descr': descr,
            }

            value = self._env.get(val_name, val_default)
            if value is None:
                if required is True:
                    raise ConfigError("%s is required" % val_name)
                else:
                    setattr(self, key, None)
            else:
                v: Any = self._get_val(val_type)
                setattr(self, key, v(val_name, value, **val)())

    @classmethod
    def _get_val(cls, val_type: Any) -> Type[Val]:
        if val_type == str:
            return StrVal
        elif val_type == bool:
            return BoolVal
        elif val_type == int:
            return IntVal
        elif val_type == float:
            return FloatVal
        elif val_type == 'file':
            return FileVal
        elif val_type == 'dir':
            return DirVal
        elif isinstance(val_type, type) and issubclass(val_type, Val):
            return val_type
        else:
            raise UserWarning('Invalid configuration settings')

    @classmethod
    def as_markdown(cls):
        result = []

        for name, options in cls._vars.items():
            options = copy.deepcopy(options)

            type_cls = cls._get_val(options.pop('type'))
            env_name = options.pop('name')
            descr = ''
            if 'descr' in options:
                descr = ': %s' % options.pop('descr')
            text = '* %s%s\n  type: %s' % (env_name, descr,
                                           type_cls.type_name())
            if 'required' in options:
                required = options.pop('required')
                if required:
                    text += '\n\n  required'
            if 'default' in options:
                default = options.pop('default')
                if default:
                    text += '\n  default: %s' % default

            inst = type_cls(env_name, None, **options)
            text += inst.args_markdown()

            result.append(text)

        return '\n\n'.join(result) + '\n'
