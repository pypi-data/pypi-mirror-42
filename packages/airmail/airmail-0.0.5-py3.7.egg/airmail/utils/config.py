from functools import reduce

def build_config(transforms, config):
    """Reduces a dictionary through functions to modify the dictionary"""
    return reduce(lambda c, fn: fn(c), transforms, config)
