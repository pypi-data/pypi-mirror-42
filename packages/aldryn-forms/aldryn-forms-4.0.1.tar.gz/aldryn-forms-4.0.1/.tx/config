[main]
host = https://www.transifex.com

[aldryn-forms.aldryn_forms]
file_filter = aldryn_forms/locale/<lang>/LC_MESSAGES/django.po
source_file = aldryn_forms/locale/en/LC_MESSAGES/django.po
source_lang = en
type = PO

