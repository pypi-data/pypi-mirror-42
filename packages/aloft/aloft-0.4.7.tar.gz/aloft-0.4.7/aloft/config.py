from aloft import input


def get_profile():
    return get_aloft_config()['profile']


def get_roles():
    return get_aloft_config()['roles']


def get_aloft_config():
    return input.read_yaml_file(get_aloft_config_filename())


def get_aloft_config_filename():
    aloft_config_directory = get_aloft_config_directory()
    return f'{aloft_config_directory}/config.yaml'


def get_aloft_config_directory():
    return input.get_directory_from_environment('ALOFT_CONFIG', '~/.aloft')
