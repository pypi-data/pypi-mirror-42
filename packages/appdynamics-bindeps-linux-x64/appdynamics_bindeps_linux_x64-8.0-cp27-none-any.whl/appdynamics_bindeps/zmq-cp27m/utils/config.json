{
  "skip_check_zmq": false, 
  "no_libzmq_extension": true, 
  "build_ext": {}, 
  "have_sys_un_h": true, 
  "zmq_prefix": "/python-agent/bindeps/deps/libzmq/binaries/release", 
  "libzmq_extension": true, 
  "bdist_egg": {}
}