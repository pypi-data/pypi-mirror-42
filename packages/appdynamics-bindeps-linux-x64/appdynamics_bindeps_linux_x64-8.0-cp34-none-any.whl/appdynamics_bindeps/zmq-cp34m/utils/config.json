{
  "build_ext": {},
  "have_sys_un_h": true,
  "skip_check_zmq": false,
  "zmq_prefix": "/python-agent/bindeps/deps/libzmq/binaries/release",
  "libzmq_extension": true,
  "no_libzmq_extension": true,
  "bdist_egg": {}
}