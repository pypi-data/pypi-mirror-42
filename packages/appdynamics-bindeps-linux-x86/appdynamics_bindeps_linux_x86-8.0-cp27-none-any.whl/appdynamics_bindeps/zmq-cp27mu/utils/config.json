{
  "libzmq_extension": true, 
  "build_ext": {}, 
  "no_libzmq_extension": true, 
  "zmq_prefix": "/python-agent/bindeps/deps/libzmq/binaries/release", 
  "have_sys_un_h": true, 
  "bdist_egg": {}, 
  "skip_check_zmq": false
}