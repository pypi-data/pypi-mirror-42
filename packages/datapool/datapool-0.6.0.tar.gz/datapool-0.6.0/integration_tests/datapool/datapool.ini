[db]
connection_string = sqlite+pysqlite:////Users/uweschmitt/Projects/eawag_datapool/src/integration_tests/lz/.simple.db

[backup_landing_zone]
folder = 

[r]
extension = .r
executable = /usr/local/bin/R

[matlab]
extension = .m
executable = /Applications/MATLAB_R2016a.app/bin/matlab

[julia]
extension = .jl
version = 0.5.2
executable = /usr/local/bin/julia

[python]
extension = .py
executable = /Users/uweschmitt/Projects/eawag_datapool/venv/bin/python3

[logging]
config_file = ./logging_config.yaml

[log_receiver]
port = 5559

[worker]
port = 5555
count = 5

[server]
pid_file = /Users/uweschmitt/.datapool/datapool.pid

[landing_zone]
folder = /Users/uweschmitt/Projects/eawag_datapool/src/integration_tests/lz

