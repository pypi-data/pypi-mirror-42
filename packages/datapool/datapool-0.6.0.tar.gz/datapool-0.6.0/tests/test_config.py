# encoding: utf-8
from __future__ import print_function, division, absolute_import

import os

from datapool.config import MagicConfig, write_ini, read_ini


def test_config(regtest):

    data = MagicConfig()
    data.a = 1
    data.b.c = "2"
    data.b.d = 3.45

    assert data.a == 1
    assert data.b.c == "2"
    assert data.b["d"] == 3.45

    assert data["a"] == 1
    assert data["b"].c == "2"
    assert data["b"]["c"] == "2"
    assert data["b"].d == 3.45
    assert data["b"]["d"] == 3.45
    data.print_(fh=regtest)
    print(data, file=regtest)


def test_config_io(tmpdir):

    data = MagicConfig()

    data.workers.queue_port = 5555
    data.workers.logger_port = 5556
    data.workers.workers_port = 5557
    data.logging.console_level = "debug"
    data.logging.file_level = "debug"
    data.logging.file_path = "~/datapool/log.txt"

    path = tmpdir.join("config.ini").strpath
    write_ini(data, path)

    data_back = read_ini(path)

    assert data_back.__file__ == path
    del data_back["__file__"]  # else next test fails
    del data["__file__"]  # else next test fails

    # test roundtrip:
    assert data == data_back


def test_config_variable_resolution(tmpdir, regtest):

    data = MagicConfig()
    data.test.tmp = "$TMP"
    data.test.home = "$HOME"

    path = tmpdir.join("config.ini").strpath
    write_ini(data, path)
    print(open(path).read(), file=regtest)

    data_back = read_ini(path)
    assert "$" not in data_back.test.tmp
    assert data_back.test.home == os.environ.get("HOME")
