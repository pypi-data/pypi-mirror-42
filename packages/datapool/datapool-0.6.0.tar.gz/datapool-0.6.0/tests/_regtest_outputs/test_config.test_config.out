 a: 1
 b:
     c: 2
     d: 3.45
 a: 1
 b:
     c: 2
     d: 3.45

